rkt/api_service.go

```diff
@@ -0,0 +1,559 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/spf13/cobra"
+	"github.com/coreos/rkt/Godeps/_workspace/src/golang.org/x/net/context"
+	"github.com/coreos/rkt/Godeps/_workspace/src/google.golang.org/grpc"
+	"github.com/coreos/rkt/api/v1alpha"
+	"github.com/coreos/rkt/common"
+	"github.com/coreos/rkt/store"
+	"github.com/coreos/rkt/version"
+)
+
+var (
+	supportedAPIVersion = "1.0.0-alpha"
+	cmdAPIService       = &cobra.Command{
+		Use:   "api-service [--listen-client-url=localhost:15441]",
+		Short: "Run API service",
+		Run:   runWrapper(runAPIService),
+	}
+
+	flagAPIServiceListenClientURL string
+)
+
+func init() {
+	cmdRkt.AddCommand(cmdAPIService)
+	cmdAPIService.Flags().StringVar(&flagAPIServiceListenClientURL, "--listen-client-url", common.APIServiceListenClientURL, "address to listen on client API requests")
+}
+
+// v1AlphaAPIServer implements v1Alpha.APIServer interface.
+type v1AlphaAPIServer struct {
+	store *store.Store
+}
+
+var _ v1alpha.PublicAPIServer = &v1AlphaAPIServer{}
```

Comment:
@alban This will let the compiler to make sure the struct implements the interface.